lib/cork/board.rb

```diff
@@ -89,9 +140,196 @@ def labeled(label, value, justification = 12)
     end
 
+    # Prints a title taking an optional verbose prefix and
+    # a relative indentation valid for the UI action in the passed
+    # block.
+    #
+    # In verbose mode titles are printed with a color according
+    # to their level. In normal mode titles are printed only if
+    # they have nesting level smaller than 2.
+    #
+    # @todo Refactor to title (for always visible titles like search)
+    #       and sections (titles that represent collapsible sections).
+    #
+    # @param [String] title
+    #        The title to print
+    #
+    # @param [String] verbose_prefix
+    #        See #message
+    #
+    # @param [FixNum] relative_indentation
+    #        The indentation level relative to the current,
+    #        when the message is printed.
+    #
+    def section(title, verbose_prefix = '', relative_indentation = 0,
+                titled = false)
+      if verbose?
+        title(title, verbose_prefix, relative_indentation)
+      elsif title_level < 1 || titled
+        puts title
+      end
+
+      @indentation_level += relative_indentation
+      @title_level += 1
+      yield if block_given?
+      @indentation_level -= relative_indentation
+      @title_level -= 1
+    end
+
+    # Prints an info to the user. The info is always displayed.
+    # It respects the current indentation level only in verbose
+    # mode.
+    #
+    # Any title printed in the optional block is treated as a message.
+    #
+    # @param [String] message
+    #        The message to print.
+    #
+    def info(message)
+      indentation = verbose? ? @indentation_level : 0
+      indented = wrap_string(message, indentation)
+      puts(indented)
+
+      if block_given?
+        @indentation_level += 2
+        @treat_titles_as_messages = true
+        yield
+        @treat_titles_as_messages = false
+        @indentation_level -= 2
+      end
+    end
+
+    # A title opposed to a section is always visible
+    #
+    # @param [String] title
+    #         The title to print
+    #
+    # @param [String] verbose_prefix
+    #         See #message
+    #
+    # @param [FixNum] relative_indentation
+    #         The indentation level relative to the current,
+    #         when the message is printed.
+    #
+    def title(title, verbose_prefix = '', relative_indentation = 2)
+      if @treat_titles_as_messages
+        message(title, verbose_prefix)
+      else
+        puts_title(title, verbose_prefix)
+      end
+
+      if block_given?
+        @indentation_level += relative_indentation
+        @title_level += 1
+        yield
+        @indentation_level -= relative_indentation
+        @title_level -= 1
+      end
+    end
+
+    # Prints a verbose message taking an optional verbose prefix and
+    # a relatvie indentation valid for the UI action in the passed block.
+    #
+    def notice(message)
+      line = "\n[!] #{message}"
+      line = line.green if ansi?
+      puts(line)
+    end
+
+    # @todo Clean interface.
```

Comment:
@segiddins - I took it from here https://github.com/CocoaPods/CocoaPods/blob/master/lib/cocoapods/user_interface.rb
https://www.dropbox.com/s/rgt5xs70jlgmj5m/Screenshot%202016-01-06%2011.18.52.png?dl=0